CMakeLists.txt

```diff
@@ -142,42 +121,31 @@ IF( WIN32 )
   set( RPCRT4 rpcrt4 )
 
   #boost
-  SET(BOOST_ROOT $ENV{BOOST_ROOT})
-#  set(Boost_USE_DEBUG_PYTHON ON)
+  if ($ENV{BOOST_ROOT})
+    SET(BOOST_ROOT $ENV{BOOST_ROOT})
+  endif()
+
+  set(Boost_USE_DEBUG_PYTHON ON)
   set(Boost_USE_MULTITHREADED ON)
   set(BOOST_ALL_DYN_LINK OFF) # force dynamic linking for all libraries
 
-  FIND_PACKAGE(Boost 1.53 REQUIRED COMPONENTS ${BOOST_COMPONENTS})
-  # For Boost 1.53 on windows, coroutine was not in BOOST_LIBRARYDIR and do not need it to build,  but if boost versin >= 1.54, find coroutine otherwise will cause link errors
-  IF(NOT "${Boost_VERSION}" MATCHES "1.53(.*)")
-    SET(BOOST_LIBRARIES_TEMP ${Boost_LIBRARIES})
-    FIND_PACKAGE(Boost 1.54 REQUIRED COMPONENTS coroutine)
-    LIST(APPEND BOOST_COMPONENTS coroutine)
-    SET(Boost_LIBRARIES ${BOOST_LIBRARIES_TEMP} ${Boost_LIBRARIES})
-  ENDIF()
+  LIST(APPEND PLATFORM_SPECIFIC_LIBS ws2_32 crypt32 mswsock userenv)
 
-  LIST(APPEND PLATFORM_SPECIFIC_LIBS wsock32.lib ws2_32.lib userenv.lib)
   # iphlpapi.lib
-
-ELSE(WIN32)
-  MESSAGE(STATUS "Configuring fc to build on Unix/Apple")
-
-  LIST(APPEND BOOST_COMPONENTS coroutine)
-
-  FIND_PACKAGE(Boost 1.53 REQUIRED COMPONENTS ${BOOST_COMPONENTS})
-
-  SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a;.so")
-
-  IF(NOT APPLE)
-    # Linux or other unix
-    SET(rt_library rt )
-    SET(pthread_library pthread)
-  ENDIF(NOT APPLE)
 ENDIF(WIN32)
 
+FIND_PACKAGE(Boost CONFIG COMPONENTS ${BOOST_COMPONENTS})
+
+IF(NOT WIN32)
+    MESSAGE(STATUS "Configuring fc to build on Unix/Apple")
 
+    if(NOT APPLE)
+        SET(rt_library rt)
+        SET(pthread_library pthread)
```

Comment:
I don't see `pthread_library` being used anywhere, is this still required?

Reply:
IIRC the `pthread_library` is not used in our project's codebase, but somehow it changed the CMake's behaviour and linking problems occurred.

I will try to build the project without the offending line.

Reply:
Looks like the line is redundant, removed.